File: gymanager/lib/src/models/trainer_model.dart
class Trainer {
  final String id;
  final String name;
  final String gender;

  Trainer({
    required this.id,
    required this.name,
    required this.gender,
  });

  factory Trainer.fromJson(Map<String, dynamic> json) {
    return Trainer(
      id: json['_id'],
      name: json['name'],
      gender: json['gender'],
    );
  }
}

File: backend_server/models/trainer.model.js
const mongoose = require('mongoose');

const trainerSchema = mongoose.Schema(
  {
    // Add this block to associate a trainer with a user
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: 'User',
    },
    name: {
      type: String,
      required: true,
    },
    gender: {
      type: String,
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

const Trainer = mongoose.model('Trainer', trainerSchema);
module.exports = Trainer;